### PR TITLE
8344187: Remove SecurityManager and related calls from java.instrument

### PR DESCRIPTION
**Pull request by kevinjwalls**
Remove redundant SecurityManager, AccessController references
(following on from JDK-8338411: Implement JEP 486: Permanently Disable the Security Manager).

test/jdk/java/lang/instrument/ still passing.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344187](https://bugs.openjdk.org/browse/JDK-8344187): Remove SecurityManager and related calls from java.instrument (**Enhancement** - P4)

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Alex Menkov](https://openjdk.org/census#amenkov) (@alexmenkov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22106/head:pull/22106` \
`$ git checkout pull/22106`

Update a local copy of the PR: \
`$ git checkout pull/22106` \
`$ git pull https://git.openjdk.org/jdk.git pull/22106/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22106`

View PR using the GUI difftool: \
`$ git pr show -t 22106`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22106.diff">https://git.openjdk.org/jdk/pull/22106.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22106#issuecomment-2479618492)
</details>

### CODE REVIEW

File: src/java.instrument/share/classes/sun/instrument/InstrumentationImpl.java [hunk 1]
```diff
@@ -485,13 +480,8 @@ private TransformerManager findTransformerManager(ClassFileTransformer transform
...
     // Enable or disable Java programming language access checks on a
     // reflected object (for example, a method)
-    @SuppressWarnings("removal")
     private static void setAccessible(final AccessibleObject ao, final boolean accessible) {
-        AccessController.doPrivileged(new PrivilegedAction<Object>() {
-                public Object run() {
-                    ao.setAccessible(accessible);
-                    return null;
-                }});
+        ao.setAccessible(accessible);
```
Comment (AlanBateman) [inline src/java.instrument/share/classes/sun/instrument/InstrumentationImpl.java:484]: I think you can remove this method, and instead change the usage in loadClassAndStartAgent to just call setAccessible.

Comment (kevinjwalls) [reply src/java.instrument/share/classes/sun/instrument/InstrumentationImpl.java:484]: will do, yes it is trivial now.

File: src/java.instrument/share/classes/sun/instrument/InstrumentationImpl.java [hunk 2]
```diff
@@ -677,11 +658,7 @@ private static URL codeSource(Class<?> clazz) {
     private static class HolderStackWalker {
         static final StackWalker walker;
         static {
-            PrivilegedAction<StackWalker> pa = () ->
-                    StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE);
-            @SuppressWarnings("removal")
-            StackWalker w = AccessController.doPrivileged(pa);
-            walker = w;
+            walker = StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE);
```
Comment (AlanBateman) [inline src/java.instrument/share/classes/sun/instrument/InstrumentationImpl.java:661]: If you want, you could remove the static block, but what you have is okay too.

**Review by AlanBateman — APPROVED** [review]

**Review by AlanBateman — APPROVED** [review]

**Review by alexmenkov — APPROVED** [review]

File: src/java.instrument/share/classes/sun/instrument/InstrumentationImpl.java [hunk 3]
```diff
@@ -36,7 +36,6 @@
 import java.nio.file.Path;
 import java.nio.file.InvalidPathException;
 import java.net.URL;
-import java.security.AccessController;
 import java.security.CodeSource;
 import java.security.PrivilegedAction;
```
Comment (eirbjo) [inline src/java.instrument/share/classes/sun/instrument/InstrumentationImpl.java:40]: A note for future cleanups in this area: This `java.security.PrivilegedAction` import should probably have been removed.

**Pull request closed without merging**